src/main/java/com/google/summit/translation/Translate.kt

```diff
@@ -1310,17 +1312,301 @@ class Translate(val file: String, private val tokens: TokenStream) : ApexParserB
   }
 
   /** Translates the 'primary#soqlPrimary' grammar rule and returns an AST [Expression]. */
```

Comment:
I know this predates this change, but I'm not quite sure what this # syntax (primary#soqlPrimary) refers to (here and elsewhere).

Reply:
It mirrors the syntax in the antlr grammar. In this case, [here](https://github.com/nawforce/apex-parser/blob/master/antlr/ApexParser.g4#L471).

Practically, the addition of the `#` seems to split off part of the rule into a named sub-rule (with its own context type and visitor method, etc.)